### PR TITLE
Sendstring LUT improvements

### PR DESCRIPTION
**Pull request by fauxpark**
## Description

 - Aligned keycode LUTs to 9 characters wide for easier reading
 - Changed `0` to `XXXXXXX` for consistency
 - Commented the keycode LUTs with the ASCII table
 - Added missing AltGr tables - those characters should now work in `SEND_STRING()` 🎉

Where possible I also changed the keycodes to their layout-specific aliases.

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: quantum/keymap_extras/sendstring_bepo.h [hunk 1]
```diff
@@ -13,29 +13,85 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-/* Sendstring definitions for the bépo layout */
-#ifndef SENDSTRING_BEPO
```
Comment (drashna) [inline quantum/keymap_extras/sendstring_bepo.h]: Honestly, this is one of the places that I think we shouldn't use `#pragma once`, so that detection can happen.

File: quantum/quantum.c [hunk 2]
```diff
@@ -837,11 +837,12 @@ bool process_record_quantum(keyrecord_t *record) {
 }
 
 __attribute__ ((weak))
-const bool ascii_to_shift_lut[0x80] PROGMEM = {
+const bool ascii_to_shift_lut[128] PROGMEM = {
```
Comment (drashna) [inline quantum/quantum.c]: Why the change from 0x80 to 128?

File: quantum/keymap_extras/sendstring_bepo.h (hunk 1 above)
Comment (fauxpark) [reply quantum/keymap_extras/sendstring_bepo.h]: What would be a common use case for it? You're probably right, but I can't think of one.

File: quantum/quantum.c (hunk 2 above)
Comment (fauxpark) [reply quantum/quantum.c]: I just prefer decimal in array lengths.

Comment (yanfali) [reply quantum/quantum.c]: If anything this could be a define since it's a magic number

Comment (fauxpark) [reply quantum/quantum.c]: Some layouts might want to map extended ASCII characters like £ (UK layout) so having a variable LUT length could be nice. Does the weak attribute allow for different array lengths here?

Comment (fauxpark) [reply quantum/quantum.c]: GCC doesn't complain when I omit the length from the LUT declarations in quantum.h, though I'm wary of what would happen if sendstring tried to look up an extended ASCII code in the default shift/altgr tables (some layouts don't need them).

File: quantum/keymap_extras/sendstring_bepo.h (hunk 1 above)
Comment (drashna) [reply quantum/keymap_extras/sendstring_bepo.h]: Just thinking about it, yeah, you're right.

File: quantum/quantum.c (hunk 2 above)
Comment (drashna) [reply quantum/quantum.c]: The better question would be if it worked.  

Also, if the shift/alt tables handle that, then do we need to extend it?

Comment (fauxpark) [reply quantum/quantum.c]: It seems that `send_char` doesn't do any sanity checks on the LUT indexes:

https://github.com/qmk/qmk_firmware/blob/7dcf9237d5cdf83e6f357755c19c14e92e6593c0/quantum/quantum.c#L837-L839

So I'm guessing you can already persuade `ascii_code` to be 129, and... I don't know what happens then. It reads some other byte next to the LUT, maybe.

Comment (drashna) [reply quantum/quantum.c]: Probably not.  If it's outside of the array, it should error out.  But you would want to make sure the array is set correctly, in that case. 

At least, IIRC.
